packages/composite-checkout/src/components/billing-fields.js

```diff
@@ -0,0 +1,528 @@
+	fieldType: PropTypes.string.isRequired,
+};
+
+function isZipOrPostal() {
+	//TODO: Add location detection to return "zip" or "postal"
+	return 'postal';
+}
+
+function DomainFields( { paymentData, setPaymentData } ) {
+	const localize = useLocalize();
+
+	return (
+		<DomainContactFields>
+			<DomainContactFieldsTitle>
+				{ localize( 'Enter your domain registration contact information' ) }
+			</DomainContactFieldsTitle>
+			<DomainContactFieldsDescription>
+				{ localize( `Domain owners have to share contact information in a public database of all domains. With
+				our Privacy Protection, we publish our own information and privately forward any
+				communication to you.` ) }
+			</DomainContactFieldsDescription>
+
+			<AddressFields
+				paymentData={ paymentData }
+				setPaymentData={ setPaymentData }
+				fieldType={ 'domains' }
+			/>
+			<TaxFields
+				paymentData={ paymentData }
+				setPaymentData={ setPaymentData }
+				fieldType={ 'domains' }
+			/>
+			<PhoneNumberField
+				paymentData={ paymentData }
+				setPaymentData={ setPaymentData }
+				fieldType={ 'domains' }
+			/>
+		</DomainContactFields>
+	);
+}
+
+DomainFields.propTypes = {
+	paymentData: PropTypes.object.isRequired,
+	setPaymentData: PropTypes.func.isRequired,
+};
+
+const DomainContactFields = styled.div`
+	margin: 16px 0 24px;
+`;
+
+const DomainContactFieldsTitle = styled.h2`
+	font-size: 16px;
+	margin: 0 0 4px;
+	font-weight: 600;
+	color: ${props => props.theme.colors.borderColorDark};
+	padding-top: 24px;
+	border-top: 1px solid ${props => props.theme.colors.borderColorLight};
+`;
+
+const DomainContactFieldsDescription = styled.p`
+	font-size: 14px;
+	color: ${props => props.theme.colors.textColor};
+	margin: 0 0 16px;
+`;
+
+function BillingFormSummary( { paymentData, isDomainContactVisible } ) {
+	const localize = useLocalize();
+	const { billing, domains } = paymentData;
+
+	//Check if paymentData is empty
+	if ( Object.entries( paymentData ).length === 0 ) {
+		return null;
+	}
+
+	return (
+		<GridRow gap="4%" columnWidths="48% 48%">
+			<div>
+				<BillingSummaryDetails>
+					<BillingSummaryLine>{ billing.name } </BillingSummaryLine>
+					<BillingSummaryLine>{ billing.address } </BillingSummaryLine>
+					<BillingSummaryLine>
+						{ billing.city }, { billing.state || billing.province }
+					</BillingSummaryLine>
+					<BillingSummaryLine>
+						{ `${ billing.zipCode || billing.postalCode }, ${ billing.country }` }
+					</BillingSummaryLine>
+				</BillingSummaryDetails>
+				{ ( billing.phoneNumber || ( isElligibleForVat() && billing.vatId ) ) && (
+					<BillingSummaryDetails>
+						<BillingSummaryLine>{ billing.phoneNumber }</BillingSummaryLine>
+						{ isElligibleForVat() && (
+							<BillingSummaryLine>
+								{ localize( 'VAT indentification number:' ) }
+								{ billing.vatId }
+							</BillingSummaryLine>
+						) }
+					</BillingSummaryDetails>
+				) }
+			</div>
+
+			{ isDomainContactVisible && (
```

Comment:
Instead of `isDomainContactVisible`, could we derive this based on the data? Like, show it if `domains.name` is set? Alternatively, instead of `isDomainContactVisible` we could have a variable in `paymentData` that's called `isDomainInfoSameAsBilling` which is basically the same thing but not directly related to the UI so it fits better in `paymentData`. Either way we'd be able to remove `isDomainContactVisible` from `Checkout`.

Reply:
> Instead of isDomainContactVisible, could we derive this based on the data?

I initially set out in this direction but the problem was that if you unchecked the box and had filled out the data, then the data would show in the summary. 

> Alternatively, instead of isDomainContactVisible we could have a variable in paymentData that's called isDomainInfoSameAsBilling which is basically the same thing but not directly related to the UI so it fits better in paymentData

This feels a little weird to me because we're doing it for the UI and it dirties up the data. Instead, I've opted to go with deriving the state based on the data and deleting any domain data when someone hides the domain fields again — this way, the domain data doesn't leak into the summary. Let me know what you think of that approach.